doc/src/manual/arrays.md

```diff
@@ -1028,33 +1028,57 @@ It is sometimes useful to perform element-by-element binary operations on arrays
 sizes, such as adding a vector to each column of a matrix. An inefficient way to do this would
 be to replicate the vector to the size of the matrix:
 
-```julia-repl
-julia> a = rand(2,1); A = rand(2,3);
+```jldoctest broadcasting
+julia> a = [1, 2]; B = [10 20 30; 40 50 60]
 
-julia> repeat(a,1,3)+A
-2×3 Array{Float64,2}:
- 1.20813  1.82068  1.25387
- 1.56851  1.86401  1.67846
+julia> repeat(a, 1, 3) + B
+2×3 Matrix{Int64}:
+ 11  21  31
+ 42  52  62
 ```
 
-This is wasteful when dimensions get large, so Julia provides [`broadcast`](@ref), which expands
-singleton dimensions in array arguments to match the corresponding dimension in the other array
-without using extra memory, and applies the given function elementwise:
+This is wasteful when dimensions get large, so Julia provides
+[`broadcast`](@ref), which aligns dimensions in array arguments starting with
+the leading ones[^2], fills lacking dimensions with a singleton dimension,
+expands singleton dimensions to match the corresponding dimension in the other
+array without using extra memory, and applies the given function elementwise:
```

Comment:
I wonder if you should leave this text alone, as it's not so easy to digest before you know what it's saying. The details it provides are important but perhaps better conveyed later.

So show the examples first, _then_ explain (just _below_ the table) that it's the first dimensions of each argument which line up. Then mention numpy. 

So many comments.